File: src/main/java/com/github/kindrat/cassandra/client/ui/eventhandler/TableClickEventHandler.java
package com.github.kindrat.cassandra.client.ui.eventhandler;

import javafx.event.EventHandler;
import javafx.scene.control.TablePosition;
import javafx.scene.control.TableView;
import javafx.scene.input.MouseEvent;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public class TableClickEventHandler<T> implements EventHandler<MouseEvent> {
    private final TableView<T> tableView;

    @Override
    public void handle(MouseEvent event) {
        //noinspection unchecked
        TablePosition<T, ?> tablePosition = tableView.focusModelProperty()
                .get()
                .focusedCellProperty()
                .get();
        tableView.edit(tablePosition.getRow(), tablePosition.getTableColumn());
    }
}

File: src/main/kotlin/com/github/kindrat/cassandra/client/ui/eventhandler/TableClickEventHandler.kt
package com.github.kindrat.cassandra.client.ui.eventhandler

import javafx.event.EventHandler
import javafx.scene.control.TableColumn
import javafx.scene.control.TableView
import javafx.scene.input.MouseEvent

class TableClickEventHandler<T>(private val tableView: TableView<T>) : EventHandler<MouseEvent?> {

    override fun handle(event: MouseEvent?) {
        val tableViewFocusModel = tableView.focusModelProperty().get()
        val tablePosition  = tableViewFocusModel.focusedCellProperty().get()

        @Suppress("UNCHECKED_CAST")
        val tableColumn: TableColumn<T, *>? = tablePosition.tableColumn as TableColumn<T, *>?
        if (tableColumn != null) {
            tableView.edit(tablePosition.row, tableColumn)
        } else {

        }
    }
}